addons/website_blog/views/website_blog_components.xml

```diff
@@ -21,9 +22,27 @@
                     </a>
                 </li>
             </ul>
+            <!-- Mobile -->
+            <div class="dropdown order-2 d-lg-none">
+                <a class="btn btn-light dropdown-toggle d-flex align-items-center" type="button" role="button" data-bs-toggle="dropdown" aria-expanded="false">
+                    <span class="d-inline-block text-truncate" t-if="blog" t-out="blog.name"/>
+                    <span class="d-inline-block text-truncate" t-else="">All</span>
+                </a>
+                <ul class="dropdown-menu dropdown-menu-end" t-if="len(blogs) > 1">
+                    <li>
+                        <a href="/blog" t-attf-class="dropdown-item #{(not blog) and 'active'}">All</a>
+                    </li>
+                    <li t-foreach="blogs" t-as="nav_blog">
+                        <a t-attf-href="/blog/#{slug(nav_blog)}"
+                        t-attf-class="dropdown-item #{(blog and nav_blog.id == blog.id) and 'active' or ''}">
+                            <span t-field="nav_blog.name"/>
+                        </a>
+                    </li>
+                </ul>
+            </div>
             <!-- 'Search Box' -->
             <t t-call="website.website_search_box_input">
-                <t t-set="_classes" t-valuef="w-100 w-md-auto pt-2 pt-md-0 d-md-flex"/>
+                <t t-set="_classes" t-valuef="w-auto d-md-flex pt-md-0 pe-3 pe-md-0"/>
```

Comment:
You traded classes order conventions for diff. readability.

Ok to me 👍
(but we should merge soon,  @Brieuc-brd is coming back from his holidays 🤫😅)